class.jetpack-user-agent.php

```diff
@@ -289,223 +55,52 @@ function get_mobile_user_agent_name() {
-			$this->isTierIphone        = true;
-			$this->isTierRichCss       = false;
-			$this->isTierGenericMobile = false;
-		} elseif ( $this->is_kindle_touch() ) {
-			$this->matched_agent       = 'kindle-touch';
-			$this->isTierIphone        = true;
-			$this->isTierRichCss       = false;
-			$this->isTierGenericMobile = false;
-		} elseif ( $this->is_Nintendo_3DS() ) {
-			$this->matched_agent       = 'nintendo-3ds';
-			$this->isTierIphone        = true;
-			$this->isTierRichCss       = false;
-			$this->isTierGenericMobile = false;
-		} else {
-			$this->isTierIphone = false;
-		}
-		return $this->isTierIphone;
+	public function isTierIphone() {
+		_deprecated_function( __METHOD__, 'Jetpack 8.7', '\Automattic\Jetpack\Device_Detection\User_Agent_Info->isTierIphone from the `automattic/jetpack-device-detection` package' );
+		return $this->ua_info->isTierIphone();
 	}
 
-	/*
+	/**
 	 * This method detects for UA which are likely to be capable
 	 * but may not necessarily support JavaScript.
 	 * Excludes all iPhone Tier UA.
-	 *
 	 */
-	function isTierRichCss() {
-		if ( isset( $this->isTierRichCss ) ) {
-			return $this->isTierRichCss;
-		}
-		if ( $this->isTierIphone() ) {
-			return false;
-		}
-
-		// The following devices are explicitly ok.
-		if ( $this->is_S60_OSSBrowser() ) {
-			$this->matched_agent       = 'series60';
-			$this->isTierIphone        = false;
-			$this->isTierRichCss       = true;
-			$this->isTierGenericMobile = false;
-		} elseif ( $this->is_opera_mini() ) {
-			$this->matched_agent       = 'opera-mini';
-			$this->isTierIphone        = false;
-			$this->isTierRichCss       = true;
-			$this->isTierGenericMobile = false;
-		} elseif ( $this->is_blackbeberry() ) {
-			$detectedDevice = $this->detect_blackberry_browser_version();
-			if ( $detectedDevice === 'blackberry-5' || $detectedDevice == 'blackberry-4.7' || $detectedDevice === 'blackberry-4.6' ) {
-				$this->matched_agent       = $detectedDevice;
-				$this->isTierIphone        = false;
-				$this->isTierRichCss       = true;
-				$this->isTierGenericMobile = false;
-			}
-		} else {
-			$this->isTierRichCss = false;
-		}
-
-		return $this->isTierRichCss;
+	public function isTierRichCss() {
+		_deprecated_function( __METHOD__, 'Jetpack 8.7', '\Automattic\Jetpack\Device_Detection\User_Agent_Info->isTierRichCss from the `automattic/jetpack-device-detection` package' );
+		return $this->ua_info->isTierRichCss();
 	}
 
-	// Detects if the user is using a tablet.
-	// props Corey Gilmore, BGR.com
-	static function is_tablet() {
-		return ( 0 // never true, but makes it easier to manage our list of tablet conditions
-				|| self::is_ipad()
-				|| self::is_android_tablet()
-				|| self::is_blackberry_tablet()
-				|| self::is_kindle_fire()
-				|| self::is_MaemoTablet()
-				|| self::is_TouchPad()
-		);
+	/**
+	 * Detects if the user is using a tablet.
+	 * props Corey Gilmore, BGR.com
+	 *
+	 * @return bool
+	 */
+	public function is_tablet() {
```

Comment:
This change from static is throwing fatals for sites running the o2 theme https://geto2.com/

Reply:
Follow-up: #16430